lib/spack/spack/test/concretize_preferences.py

```diff
@@ -228,6 +211,37 @@ def test_external_mpi(self):
         spec.concretize()
         assert spec['mpich'].external_path == '/dummy/path'
 
+    def test_external_module(self, save_module_func):
+        """Test that packages can find externals specified by module
+
+        The specific code for parsing the module is tested elsewhere.
+        This just tests that the preference is accounted for"""
+        # make sure this doesn't give us an external first.
+        def mock_module(cmd, module):
+            return 'prepend-path PATH /dummy/path'
+        spack.util.module_cmd.module = mock_module
```

Comment:
Wondering if this:
```suggestion
    def test_external_module(self, monkeypatch):
        """Test that packages can find externals specified by module

        The specific code for parsing the module is tested elsewhere.
        This just tests that the preference is accounted for"""
        # make sure this doesn't give us an external first.
        def mock_module(cmd, module):
            return 'prepend-path PATH /dummy/path'
        monkeypatch.setattr(spack.util.module_cmd, 'module', mock_module)
```
works, so that we can remove the new fixture in `conftest.py`